### PR TITLE
ROX-24944: Improve crdwatcher unit test

### PR DESCRIPTION
**Pull request by lvalerom**
### Description

Wait for resources to be created before asserting to avoid possible races.

### User-facing documentation

- [x] CHANGELOG update is not needed
- [x] Documentation is not needed

### Testing

- [x] inspected CI results

#### Automated testing

- [x] modified existing tests
  

#### How I validated my change

I've run the tests multiple times locally.

### CODE REVIEW

File: sensor/kubernetes/listener/watcher/crd/crdwatcher_test.go [hunk 1]
```diff
@@ -66,20 +66,43 @@ func newFakeCRD(name string) *unstructured.Unstructured {
...
+		err := s.dynamicClient.Resource(gvr).Delete(context.Background(), name, metav1.DeleteOptions{})
+		s.Require().NoError(err)
 	}
 }
 
+func (s *watcherSuite) waitForResourcesCreation(resources ...string) {
+	s.Eventually(func() bool {
+		// The context is ignored in the fake client, so we can simply pass context.Background
+		list, err := s.dynamicClient.Resource(gvr).List(context.Background(), metav1.ListOptions{})
+		if err != nil || len(list.Items) != len(resources) {
```
Comment (vikin91) [inline sensor/kubernetes/listener/watcher/crd/crdwatcher_test.go:89]: Fully optional
```suggestion
		return err == nil && len(list.Items) == len(resources)
```

File: sensor/kubernetes/listener/watcher/crd/crdwatcher_test.go [hunk 2]
```diff
@@ -66,20 +66,43 @@ func newFakeCRD(name string) *unstructured.Unstructured {
...
+	}, defaultTimeout, time.Millisecond, "the expected resources were not created on time: %v", resources)
+}
+
+func (s *watcherSuite) waitForResourcesRemoval() {
+	s.Eventually(func() bool {
+		list, err := s.dynamicClient.Resource(gvr).List(context.Background(), metav1.ListOptions{})
+		if err != nil || len(list.Items) > 0 {
+			return false
+		}
+		return true
```
Comment (vikin91) [inline sensor/kubernetes/listener/watcher/crd/crdwatcher_test.go:102]: Same compacting method possible here

**Review by vikin91 — APPROVED** [review]
LGTM!

File: sensor/kubernetes/listener/watcher/crd/crdwatcher_test.go (hunk 1 above)
Comment (lvalerom) [reply sensor/kubernetes/listener/watcher/crd/crdwatcher_test.go]: Done

File: sensor/kubernetes/listener/watcher/crd/crdwatcher_test.go (hunk 2 above)
Comment (lvalerom) [reply sensor/kubernetes/listener/watcher/crd/crdwatcher_test.go]: Done

**Pull request merged**